packages/aws-cdk-lib/verify-stripped-exp.js

```diff
@@ -0,0 +1,134 @@
+  process.chdir(tempDir);
+  const version = require('./package.json').version;
+  const tarFullPath = path.join(cwd, JS_DIST_LIB, `aws-cdk-lib-${version}.tgz`);
+
+  // install the tarball in a temp directory
+  exec('npm', ['init', '-y']);
+  exec('npm', ['install', tarFullPath]);
+  const installedAwsCdkLibPath = path.join('node_modules', 'aws-cdk-lib', 'lib');
+
+  for (const module of fs.readdirSync(awsCdkModulesRepoPath)) {
+    const pkgJson = require(path.join(awsCdkModulesRepoPath, module, 'package.json'));
+    if (pkgJson['stability'] !== 'experimental') {
+      continue;
+    }
+    if (pkgJson['cdk-build'].cloudformation) {
+      // if a cfn module, verify only the allowed files exists
+      const files = await getAllFiles(path.join(installedAwsCdkLibPath, module));
+      files.forEach(file => {
+        if (!isAllowedFile(file)) {
+          throw `only L1 auto generated files are allowed in experimental modules included in aws-cdk-lib, the file ${file} exists in ${module}`;
+        }
+      });
+    } else {
+      // not a cfn module, verify it was entirely removed
+      if (fs.existsSync(path.join(installedAwsCdkLibPath, module))) {
+        throw `only L1s of experimental modules are allowed in aws-cdk-lib. The higher level module ${module} was included`;
+      }
+    }
+  }
+}
+
+const tempDir = fs.mkdtempSync(os.tmpdir());
+
+main(tempDir).then(
+  () => {
+    fs.removeSync(tempDir);
+    process.exit(0);
+  },
+  (err) => {
+    process.stderr.write(`${err.toString()}\n`);
+    process.stderr.write(`did not delete the temp folder: ${tempDir}`)
+    process.exit(1);
+  },
+);
+
+
+/**
+ * Spawn sync with error handling
+ */
+function exec(cmd, args) {
+  const proc = spawnSync(cmd, args);
+
+  if (proc.error) {
+    throw proc.error;
+  }
+
+  if (proc.status !== 0) {
+    if (proc.stdout || proc.stderr) {
+      throw new Error(`[Status ${proc.status}] stdout: ${proc.stdout?.toString().trim()}\n\n\nstderr: ${proc.stderr?.toString().trim()}`);
+    }
+    throw new Error(`${cmd} exited with status ${proc.status}`);
+  }
+
+  return proc;
+}
+
+const GENERATED_SUFFIX_REGEX = new RegExp(/generated\.(js|ts|d\.ts)$/);
+const ALLOWED_FILES = ['.jsiirc.json', 'index.ts', 'index.js', 'index.d.ts'];
+
+/**
+ * Recursively collect all files in dir
+ */
+async function getAllFiles(dir) {
+  const ret = new Array();
+
+  async function recurse(dir) {
+    const files = await fs.readdir(dir);
+    for (const file of files) {
+      const fullPath = path.join(dir, file);
+      if ((await fs.stat(fullPath)).isDirectory()) {
+        await recurse(fullPath);
+      } else {
+        ret.push(file);
+      }
+    }
+  }
+  await recurse(dir);
+  return ret;
+}
+
+/**
+ * Find the workspace root path. Walk up the directory tree until you find lerna.json
+ */
+ function findWorkspacePath() {
+
+  return _findRootPath(process.cwd());
+
+  function _findRootPath(part) {
+    if (process.cwd() === os.homedir()) {
```

Comment:
Shouldn't this be:

```suggestion
    if (part === os.homedir()) {
```

?

Reply:
Also better to compare to `path.resolve('/', '..')` so that it stops if the script happens to not from run from inside the homedir (/tmp)

Reply:
Good catch! Even worse, this was copied from the original `ubergen` script, which means the bug was other, and the fix needs to be applied in both:/. We really need to have a way to share "util" function between modules. I'm going to start a thread